src/NuGet.Services.AzureSearch/Catalog2AzureSearch/AzureSearchCollectorLogic.cs

```diff
@@ -36,6 +36,20 @@ public class AzureSearchCollectorLogic : ICommitCollectorLogic
             _options = options ?? throw new ArgumentNullException(nameof(options));
             _telemetryService = telemetryService ?? throw new ArgumentNullException(nameof(telemetryService));
             _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+
+            if (_options.Value.MaxConcurrentBatches <= 0)
+            {
+                throw new ArgumentException(
```

Comment:
Super nit: `ArgumentOutOfRangeException`? Same for line 49

Reply:
Sounds good.